lib/game.rb

```diff
@@ -14,14 +14,60 @@ def self.from_csv(csv_file_path)
 
   def self.count_of_games_by_season
     games_by_season = @@all.group_by { |game| game.season }
-    number_of_games_in_season =
     count = {}
     games_by_season.keys.each do |key|
       count[key] = @@all.count { |game| game.season == key}
     end
     count
   end
 
+#one of THE METHODS ross
+  def self.average_goals_per_game
+    sum = @@all.sum { |game| game.away_goals + game.home_goals}.to_f
+    (sum / @@all.length.to_f).round(2)
+  end
+
+  def self.games_per(header)
+    group_by_header = @@all.group_by { |game| game.send(header) }
+    group_by_header.values.map{ |games| games.length}
+  end
+
+
+  def self.goals_per(header, goals)
+    group_by_header = @@all.group_by { |game| game.send(header) }
+    group_by_header.values.map do |games|
+      games.sum { |game| (game.send(goals))}
+    end
+  end
+
+  def self.home_away_goals_per(header)
+    goals_per(header, :away_goals) + goals_per(header, :home_goals)
+  end
+
+  def self.average_goals(sum_array, length_array)
+    sum_array.each_with_index.map do |goals, index|
+      (goals.to_f / length_array[index].to_f ).round(2)
+    end
+  end
+#one of THE METHODS ross
+  def self.average_goals_by_season
+    goals = home_away_goals_per(:season)
+    season_length = games_per(:season)
+    average_goals_per_season = average_goals(goals, season_length)
+    season_ids = @@all.map { |game| game.season}.uniq
+    Hash[season_ids.zip(average_goals_per_season)]
+  end
```

Comment:
This method is able to pull 4 deliverables 
highest scoring visitor
highest scoring home team
lowest scoring visitor
lowest scoring home team

Reply:
I'm sure this is all very smart but I don't understand any of it

Reply:
https://media.giphy.com/media/FmBhzktIjvdZe/giphy.gif